packages/components/src/color-picker/index.js

```diff
@@ -106,140 +113,131 @@ const dataToColors = ( oldColors, { source, valueKey, value } ) => {
 		source,
 		...{ ...oldColors[ source ], ...{ [ valueKey ]: value } },
 	};
-};
-
-export default class ColorPicker extends Component {
-	constructor( { color = '0071a1' } ) {
-		super( ...arguments );
-		const colors = colorToState( color );
-		this.state = {
-			...colors,
-			draftHex: toLowerCase( colors.hex ),
-			draftRgb: colors.rgb,
-			draftHsl: colors.hsl,
-		};
-		this.commitValues = this.commitValues.bind( this );
-		this.setDraftValues = this.setDraftValues.bind( this );
-		this.resetDraftValues = this.resetDraftValues.bind( this );
-		this.handleInputChange = this.handleInputChange.bind( this );
-	}
-
-	commitValues( data ) {
-		const { oldHue, onChangeComplete = noop } = this.props;
+}
 
+export default function ColorPicker( {
+	color = '0071a1',
+	className,
+	disableAlpha,
+	oldHue,
+	onChangeComplete = noop,
+} ) {
+	const initialColor = colorToState( color );
+	const [ colors, setColors ] = useState( {
+		...initialColor,
+		draftHex: toLowerCase( initialColor.hex ),
+		draftRgb: initialColor.rgb,
+		draftHsl: initialColor.hsl,
+	} );
+	const { hex, hsl, hsv, rgb, draftHex, draftHsl, draftRgb } = colors;
+
+	const classes = classnames( className, {
+		'components-color-picker': true,
+		'is-alpha-disabled': disableAlpha,
+		'is-alpha-enabled': ! disableAlpha,
+	} );
+
+	function commitValues( data ) {
 		if ( isValidColor( data ) ) {
-			const colors = colorToState( data, data.h || oldHue );
-			this.setState(
-				{
-					...colors,
-					draftHex: toLowerCase( colors.hex ),
-					draftHsl: colors.hsl,
-					draftRgb: colors.rgb,
-				},
-				debounce( partial( onChangeComplete, colors ), 100 )
-			);
+			const newColors = colorToState( data, data.h || oldHue );
+			setColors( {
+				...newColors,
+				draftHex: toLowerCase( newColors.hex ),
+				draftHsl: newColors.hsl,
+				draftRgb: newColors.rgb,
+			} );
+
+			debounce( partial( onChangeComplete, newColors ), 100 );
```

Comment:
This instance of `debounce` doesn't seem to be cleaned up. Maybe a good place to use the new `useDebounce`?

Reply:
I've switched to useDebounce in daddou-ma/gutenberg@1e99144